closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -362,6 +383,47 @@ int encryptFinal(byte[] in, int inOfs, int len, byte[] out, int outOfs)
         return (len + tagLenBytes);
     }
 
+    /**
+     * Performs encryption operation for the last time.
+     *
+     * @param src the input buffer with the data to be encrypted
+     * @param dst the offset in <code>in</code>
+     * @return the number of bytes placed into the <code>out</code> buffer
+     */
+    int encryptFinal(ByteBuffer src, ByteBuffer dst)
+        throws IllegalBlockSizeException, ShortBufferException {
+        byte[] dst_arr = null;
+
+        // Get array from source
+        byte[] src_arr = getBbArray(src);
+        int src_ofs = src.position() + (src.hasArray() ? src.arrayOffset() : 0);
+
+        // Get array from destination
+        if (dst.hasArray()) {
+            dst_arr = dst.array();
+        }
+        else {
+            dst_arr = new byte[dst.capacity()];
+        }
+        int dst_offset = dst.position() + (dst.hasArray() ? dst.arrayOffset() : 0);
+
+        int len = encryptFinal(src_arr, src_ofs, src.remaining(), dst_arr, dst_offset);
```

Comment:
I expect a primary motivation for supporting `ByteBuffer` is for performance of the case where it is not backed by an array (e.g. direct buffers). Allocating and copying to an array sacrifices that performance expectation.

Reply:
The Cipher API requires handling ByteBuffer inputs, which is the main motivation behind this PR. In terms of performance, the only way to send the ByteBuffer data from Java to OpenSSL is by converting it to a byte[]. Looking further at how to optimize that is beyond the scope of this PR.

Reply:
Please open an issue at https://github.com/eclipse/openj9/issues to track such improvements.